### ISSUE TITLE
$__rate_interval is broken on the docker ARM image

### ISSUE THREAD

**Issue by simphide** [grafana/grafana#43002]
**What happened**: $__rate_interval has not the correct value and the visuals using $__rate_interval don't show any data

![image](https://user-images.githubusercontent.com/8183517/145675718-53824b85-a3e2-4182-b966-df17b83678ec.png)

**What you expected to happen**: $__rate_interval is definied as max( $__interval + Scrape interval, 4 * Scrape interval). The interval is set to 1m and the scrape interval is the prometheus default (15s). so $__rate_interval should be 75s. Instead $__rate_interval is 2s (see screenshot below)
![image](https://user-images.githubusercontent.com/8183517/145675525-b3c6c45e-4949-4dc9-b421-4dd138304ec6.png)

This happend with two different raspberry pis running Grafana via docker, but the x64 image seems to work fine.

**How to reproduce it (as minimally and precisely as possible)**:
Install the Grafana docker image on a raspberry pi and try to use $__rate_interval in a visual.

**Anything else we need to know?**: No

**Environment**: 
- Grafana version: 8.3.2
- Data source type & version: Prometheus 2.31.1
- OS Grafana is installed on: Raspbian GNU/Linux 10 (buster) 
- User OS & Browser: macOS und Google Chrome 
- Grafana plugins: -
- Others: -

**Comment by Elfo404** [comment]
hi @simphide thanks for reporting this!

I was actually able to reproduce this on a rpi 4 using grafana-oss docker image. 

this is the actual query being sent:
```
{"queries":[{"expr":"$__rate_interval","legendFormat":"","interval":"","exemplar":true,"refId":"A","datasource":{"uid":"nFKgBemRz","type":"prometheus"},"queryType":"timeSeriesQuery","requestId":"2A","utcOffsetSec":0,"datasourceId":1,"intervalMs":60000,"maxDataPoints":1021}],"range":{"from":"2021-12-14T10:50:31.253Z","to":"2021-12-14T16:50:31.253Z","raw":{"from":"now-6h","to":"now"}},"from":"1639479031253","to":"1639500631253"}
```

**Comment by currybullen** [comment]
I have had this issue as well, at least since 8.3.1, running the official container image for armv7 on a RPi 4.

**Comment by JsBergbau** [comment]
It is not only the docker image. Problem exists also when installed via apt and the official repo from `deb https://packages.grafana.com/oss/deb stable main`

![grafik](https://user-images.githubusercontent.com/37013344/147006743-7f44da6e-607b-4a03-bd51-f873ad562878.png)

Please fix soon. 
Thanks for your support.

**Comment by Lef-F** [comment]
Same issue on RPi 10 (buster) 
- kernel: 5.10.63-v7l+
- Grafana: v8.3.3 (30bb7a93ca)
- prometheus: 2.32.1
- go: 1.17.5
- node_exporter: 1.3.1.linux-armv7

My current workaround is to switch to `$__interval`. 

To do it fast on a current dashboard, in Grafana go to `your dashboard > Settings > JSON model` and find & replace (ctrl+F or cmd+F) `$__rate_interval` with `$__interval` or the interval you wish e.g. `1m`.

**Comment by dbrennand** [comment]
I also have this issue on Raspberry Pi 4.

* OS: Raspbian GNU/Linux 10 (buster)
* kernel: 5.10.60-v7l+
* Grafana: grafana/grafana:8.3.3
* prometheus: prom/prometheus:v2.32.1
* node_exporter: node-exporter-linux-armv7:v1.3.1

**Comment by torkelo** [comment]
Strange, that the rate interval calculation is different on Raspberry pi? Maybe the image is old? What does the image commit say? (number after the version)?

**Comment by dbrennand** [comment]
> Strange, that the rate interval calculation is different on Raspberry pi? Maybe the image is old? What does the image commit say? (number after the version)?

v8.3.3 (30bb7a93ca)

**Comment by JsBergbau** [comment]
> $__rate_interval



> Same issue on RPi 4
> 
>     * OS: Raspbian GNU/Linux 10 (buster)
> 
>     * kernel: 5.10.63-v7l+
> 
>     * Grafana: v8.3.3 ([30bb7a9](https://github.com/grafana/grafana/commit/30bb7a93cad59b3814d4017e263d827804037ebf))
> 
>     * prometheus: 2.32.1
> 
>     * go: 1.17.5
> 
>     * node_exporter: 1.3.1.linux-armv7
> 
> 
> My current workaround is to switch to `$__interval`.
> 
> To do it fast on a current dashboard, in Grafana go to `your dashboard > Settings > JSON model` and find & replace (ctrl+F or cmd+F) `$__rate_interval` with `$__interval` or the interval you wish e.g. `1m`.
> 
> In my case this breaks parts of the dashboard [`Node Exporter Full by rfraile`](https://grafana.com/grafana/dashboards/1860) or [on GitHub](https://github.com/rfrail3/grafana-dashboards)

Thanks for the fix. However when zooming too much / short time interval no data is found with this. Thats why I replaced `$__rate_interval` with `60s`. `60s` is nowhere else at the dashboard, so you can easily replace this back to `$__rate_interval` when this problem is hopefully soon fixed. This is only a quick hack, because 60s interval is not very efficient for longer time intervals. I really hope that this is fixed soon.

**Comment by sud33p** [comment]
> > My current workaround is to switch to `$__interval`.
> > To do it fast on a current dashboard, in Grafana go to `your dashboard > Settings > JSON model` and find & replace (ctrl+F or cmd+F) `$__rate_interval` with `$__interval` or the interval you wish e.g. `1m`.
> > In my case this breaks parts of the dashboard [`Node Exporter Full by rfraile`](https://grafana.com/grafana/dashboards/1860) or [on GitHub](https://github.com/rfrail3/grafana-dashboards)
> 
> Thanks for the fix. However when zooming too much / short time interval no data is found with this. Thats why I replaced `$__rate_interval` with `60s`. `60s` is nowhere else at the dashboard, so you can easily replace this back to `$__rate_interval` when this problem is hopefully soon fixed. This is only a quick hack, because 60s interval is not very efficient for longer time intervals. I really hope that this is fixed soon.

I am facing the same issue on RPi 4 (via docker), and thanks for the quick workaround suggestions. In my case, I added a [constant variable](https://grafana.com/docs/grafana/latest/variables/variable-types/add-constant-variable/) on the dashboard called $fix_rate_interval with value `75s` and replaced `__rate_interval` with `fix_rate_interval` in the JSON model. This would make it a bit easier to change values from the dashboard settings directly. Hoping the issue gets fixed soon.

**Comment by blalop** [comment]
The issue is not present on Grafana 8.2.7. Downgrading from 8.3.3 to 8.2.7 made the fix for me until a proper fix is released.

**Comment by JsBergbau** [comment]
Could that also be an issue with Prometheus and Grafana in combination? I've switched to VictoriaMetrics, which is mainly compatible with PromQL language and `$__rate_interval` works now as expected.

**Comment by JustJoshingIT** [comment]
I've been having this problem since the beginning of this year. I just checked and a new update for Grafana (8.3.6) was released. I updated and rebooted my OS and it looks like they fixed the problem.

So if you've been running on 8.2.7 (like I have been) and have been waiting for an update to fix the problem it should be safe to update.

Details: I'm running Ubuntu DE 20.04LTS with Grafana-Enterprise stable (8.3.6), Prometheus (2.15.2 I think), with the "Node Exporter Full" Grafana dashboard. If you're using something different there is a chance the bug has not been fixed for your setup.

**Comment by jschuiteboer** [comment]
I also found my way here after seeing errors on the Node Exporter Full dashboard. Here's the full error (so google can pick it up)
> not a valid duration string: "2.147483647s"

```
Hardware: Raspberry Pi 3B+
OS: Raspbian GNU/Linux 11 (bullseye)

Docker containers:
  - prom/prometheus:v2.33.1 (with default settings)
  - grafana/grafana-oss:8.3.4 (prometheus datasource has scrape interval of 15s)
```

I tested various versions by plopping $__rate_interval in the explorer and seeing what duration is reported. Here are my findings:
8.3.6 - 2.147483647s
8.3.4 - 2.147483647s
8.3.3 - 2s
8.3.2 - 2s
8.2.7 - 60s

8.2.7 seems to be the correct behavior (4 x 15s = 60s).

**Comment by Lef-F** [comment]
Just downgraded to `8.2.7` by running `sudo apt-get install grafana=8.2.7` on my raspberry pi and I can also verify that `$__rate_interval` now works as expected.

**Comment by B1773rm4n** [comment]
Confirm. Same issue here. When I change  "$__rate_interval" to "5m" it solves the issue. So confirm what OP says.
Environment:

-     Grafana version: Grafana v8.4.5
-     Data source type & version: Prometheus
-     OS Grafana is installed on: Alpine Linux 3.15.3 (Docker container)
-     User OS & Browser: Arch, Firefox

**Comment by dominik-ba** [comment]
Grafana versions 
- 8.4.2 and 
- 8.4.7 
still have the same issue. Since it is known that 8.2.7 is working I wonder is the RC not known? No fix planned?

**Comment by unknown321** [comment]
grafana/grafana-oss:9.0.3, armv7, same problem

**Comment by manibatra** [comment]
grafana/grafana:9.0.4, arm7, same problem

**Comment by ywwg** [comment]
bumping because I'm still seeing this on raspberry pi.  Also the error message is strange, as noted above it says `not a valid duration string: "2.147483647s"` which implies that the code is unable to read the float + 's'. Too many decimal places or something??  (e.g. would 2.15s work?)  Or is it "invalid" because it's too little time?

**Comment by ywwg** [comment]
suspiciously, 2147483647 is maxint32

**Comment by ywwg** [comment]
We have narrowed down the issue. "int()" was used where "int64()" should have been used instead.  We are working on a fix and will probably start to do more thorough testing of arm+32bit to catch other similar issues

**Comment by gtk-grafana** [comment]
This was fixed in [#51508](https://github.com/grafana/grafana/pull/51508) and released in [9.1.0](https://github.com/grafana/grafana/releases/tag/v9.1.0). Please let us know if you're still having trouble on 9.1.0 or greater.
